docs/src/reference/the-traversal.asciidoc

```diff
@@ -4916,6 +4916,28 @@ g.V().hasLabel('person'). <1>
 <8> `PathRetractionStrategy` will remove paths from the traversers and increase the likelihood of bulking as path data is not required after `select('b')`.
 <9> `AdjacentToIncidentStrategy` will turn `out()` into `outE()` to increase data access locality.
 
+=== DetachStrategy
+
+`DetachStrategy` manages the fields that will be included in the result. The strategy adds a step to the end of the traversal 
+and does not affect the previous steps.
+There are 3 options for detachMode: `ALL` for all properties, `NONE` for no properties, `CUSTOM` for a specific list of properties.
+`id` and `label` always included in response.
+Same for Vertexes and Edges.
+
+WARNING: by default Gremlin Server configured to use `ReferenceElementStrategy` which removes all properties.
```

Comment:
1. What happens if `ReferenceElementStrategy` and `DetachStrategy` are both used at the same time?
2. I wonder what should happen to `ReferenceElementStrategy` in 3.7.x...is it even relevant anymore as there is `DetachStrategy` with mode `NONE` which is the same thing basically? I guess one returns a `ReferenceElement` and the other a `DetachedElement` with no properties so there is that difference. thoughts?

Reply:
1. The result will be Vertex without properties. I don't think it makes sense to use them at the same time.
2. I think `ReferenceElementStrategy` may be deprecated in 3.7.
There are some small differences between ReferenceElement and DetachedElement that need to be aligned before. For example https://github.com/apache/tinkerpop/blob/master/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/reference/ReferenceElement.java#L50 and https://github.com/apache/tinkerpop/blob/master/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/detached/DetachedElement.java#L49
3. I consider default may be `DetachStrategy` configured to skip properties. I think users who want to get properties can remove the strategy from server startup or use it from the client when needed. Most current users don't expect to get properties and it doesn't make sense to load the server and network.
Another option is to remove `ReferenceElementStrategy` and allow users to choose the settings themselves, with the recommendation to skip properties by default.

Reply:
> I think ReferenceElementStrategy may be deprecated in 3.7.
There are some small differences between ReferenceElement and DetachedElement that need to be aligned before.

I'm not sure that `ReferenceElement` can be replaced by `DetachedElement` for all use cases. The original point of `ReferenceElement` is to avoid the the memory costs of even having an empty `Map<String, List<Property>> properties` declaration at all for OLAP use cases. `ReferenceElement` is meant to be as light as possible. If it could be proven that `DetachedElement` could have that same memory footprint I suppose the entire `ReferenceElement` infrastructure could be deprecated. 

As far as `ReferenceElementStrategy` goes.......I think that can be deprecated as part of this because it is not use in OLAP iirc. It is only used for the case `DetachStrategy` is and is behaviorally identical for practical purposes.

> I consider default may be DetachStrategy configured to skip properties. I think users who want to get properties can remove the strategy from server startup or use it from the client when needed. Most current users don't expect to get properties and it doesn't make sense to load the server and network.

i'd agree. so the only real change users will get is that they will now get a `DetachedElement` rather than a `ReferenceElement` and hopefully they haven't been coding to the implementation and will not notice the difference as they would be working with the interface of `Element`.  The upgrade documentation will need to be sure to include this notice.

Reply:
I can change detach mode `NONE` or add detach mode `REFERENCE` to build `ReferenceElement`.

Reply:
that's interesting. We often refer to getting a `ReferenceElement` as "reference detachment" so that might be a neat way to centralize all detachment functions behind a single strategy. seems like a good idea imo